sql/src/main/java/io/crate/action/sql/query/CrateSearchService.java

```diff
@@ -117,6 +117,65 @@ public CrateSearchService(Settings settings,
         sortSymbolVisitor = new SortSymbolVisitor(inputSymbolVisitor);
     }
 
+
+    public ScrollQueryFetchSearchResult executeScrollPhase(QueryShardScrollRequest request) {
+        final SearchContext context = findContext(request.id());
+        contextProcessing(context);
+        try {
+            processScroll(request, context);
+            context.indexShard().searchService().onPreQueryPhase(context);
+            long time = System.nanoTime();
+            try {
+                queryPhase.execute(context);
+            } catch (Throwable e) {
+                context.indexShard().searchService().onFailedQueryPhase(context);
+                throw ExceptionsHelper.convertToRuntime(e);
```

Comment:
do we need a ElasticsearchException here? Otherwise probably better to use something like ``Throwables.propagate``